cns/middlewares/k8sSwiftV2.go

```diff
@@ -135,150 +133,71 @@ func (m *K8sSWIFTv2Middleware) validateIPConfigsRequest(ctx context.Context, req
-		if err != nil {
-			return errors.Wrapf(err, "failed to parse serviceCIDRs")
-		}
-		// Check if the podIPInfo is IPv4 or IPv6
-		ip, err := netip.ParseAddr(podIPInfo.PodIPConfig.IPAddress)
-		if err != nil {
-			return errors.Wrapf(err, "failed to parse podIPConfig IP address %s", podIPInfo.PodIPConfig.IPAddress)
-		}
-		if ip.Is4() {
-			// routes for IPv4 podCIDR traffic
-			for _, podCIDRv4 := range podCIDRsV4 {
-				podCIDRv4Route := cns.Route{
-					IPAddress:        podCIDRv4,
-					GatewayIPAddress: overlayGatewayv4,
-				}
-				podIPInfo.Routes = append(podIPInfo.Routes, podCIDRv4Route)
-			}
-			// route for IPv4 serviceCIDR traffic
-			for _, serviceCIDRv4 := range serviceCIDRsV4 {
-				serviceCIDRv4Route := cns.Route{
-					IPAddress:        serviceCIDRv4,
-					GatewayIPAddress: overlayGatewayv4,
-				}
-				podIPInfo.Routes = append(podIPInfo.Routes, serviceCIDRv4Route)
-			}
-			// route for IPv4 infraVNETCIDR traffic
-			for _, infraVNETCIDRv4 := range infraVNETCIDRsv4 {
-				infraVNETCIDRv4Route := cns.Route{
-					IPAddress:        infraVNETCIDRv4,
-					GatewayIPAddress: overlayGatewayv4,
-				}
-				podIPInfo.Routes = append(podIPInfo.Routes, infraVNETCIDRv4Route)
+			return nil, errors.Wrap(err, "failed to parse mtpnc primary IP and prefix")
+		}
+		if prefixSize != prefixLength {
+			return nil, errors.Wrapf(errInvalidMTPNCPrefixLength, "mtpnc primaryIP prefix length is %d", prefixSize)
+		}
+
+		podIPInfos = append(podIPInfos, cns.PodIpInfo{
+			PodIPConfig: cns.IPSubnet{
+				IPAddress:    ip,
+				PrefixLength: uint8(prefixSize),
+			},
+			MacAddress:        mtpnc.Status.MacAddress,
+			NICType:           cns.DelegatedVMNIC,
+			SkipDefaultRoutes: false,
+			// InterfaceName is empty for DelegatedVMNIC
+		})
+	} else {
+		// Use InterfaceInfos if not empty
+		podIPInfos = make([]cns.PodIpInfo, len(mtpnc.Status.InterfaceInfos))
+		for i, interfaceInfo := range mtpnc.Status.InterfaceInfos {
+			// Parse MTPNC primaryIP to get the IP address and prefix length
+			ip, prefixSize, err := utils.ParseIPAndPrefix(interfaceInfo.PrimaryIP)
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to parse mtpnc primary IP and prefix")
 			}
-		} else {
-			// routes for IPv6 podCIDR traffic
-			for _, podCIDRv6 := range podCIDRv6 {
-				podCIDRv6Route := cns.Route{
-					IPAddress:        podCIDRv6,
-					GatewayIPAddress: overlayGatewayV6,
-				}
-				podIPInfo.Routes = append(podIPInfo.Routes, podCIDRv6Route)
+			if prefixSize != prefixLength {
+				return nil, errors.Wrapf(errInvalidMTPNCPrefixLength, "mtpnc primaryIP prefix length is %d", prefixSize)
 			}
-			// route for IPv6 serviceCIDR traffic
-			for _, serviceCIDRv6 := range serviceCIDRsV6 {
-				serviceCIDRv6Route := cns.Route{
-					IPAddress:        serviceCIDRv6,
-					GatewayIPAddress: overlayGatewayV6,
-				}
-				podIPInfo.Routes = append(podIPInfo.Routes, serviceCIDRv6Route)
+
+			nicType := cns.DelegatedVMNIC
```

Comment:
can you check for device type (interfaceInfo.DeviceType) & assign for backendnic too? 
1. if devicetype=Vnet & accelnet=false -> delegatedvmnictype, 
2. if devicetype=Vnet & accelnet=true -> accelnetnictype
3. if devicetype=Backend/IB -> backendnictype